core/txpool/txpool.go

```diff
@@ -706,17 +731,34 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (replaced bool, err e
 			overflowedTxMeter.Mark(1)
 			return false, ErrTxPoolOverflow
 		}
+
+		// If the new transaction is a future transaction it should never churn pending transactions
+		if pool.isFuture(from, tx) {
+			for _, dropTx := range drop {
+				dropSender, _ := types.Sender(pool.signer, dropTx)
+				if list := pool.pending[dropSender]; list != nil && list.Overlaps(dropTx) {
+					// Add all transactions back to the priced queue
+					for _, d := range drop {
+						pool.priced.urgent.Push(d)
+					}
+					log.Trace("Discarding future transaction replacing pending tx", "hash", hash)
+					return false, ErrFutureReplacePending
```

Comment:
By breakig here, and not continuing the iteration, it's possible that out of e.g `4` dropped transctions, you only put back the first one

Reply:
Ah wait my bad. Can you rewrite this somehow? 
```
for x in a...b, 
   if baz 
        for o in a...b
```
This feels a bit convoluted